tfutils/base.py

```diff
@@ -97,83 +98,131 @@ def __init__(self,
+            setattr(self, _k, save_params.get(_k, DEFAULT_SAVE_PARAMS[_k]))
+
+        for _k in ['do_restore']:
+            setattr(self, _k, load_params.get(_k, DEFAULT_LOAD_PARAMS[_k]))
+
+        self.conn = pymongo.MongoClient(host=self.host, port=self.port)
+        self.collfs = gridfs.GridFS(self.conn[self.dbname], self.collname)
+        recent_name = '_'.join([self.dbname, self.collname, self.exp_id, '__RECENT'])
         self.collfs_recent = gridfs.GridFS(self.conn[recent_name])
-        self.coll_recent = self.collfs_recent._GridFS__files
 
-        if tensorboard_dir is not None:  # save graph to tensorboard
-            tf.train.SummaryWriter(tensorboard_dir, tf.get_default_graph())
+        self.load_data = None     
+        load_query = load_params.get('query')
+        if load_query is None:
+            load_query = {}
+        else:
+            if self.sameloc:
+                raise Exception('Loading pointlessly')
+        load_query.update({'exp_id': self.load_exp_id})
+        self.load_query = load_query       
+        if self.load_host != self.host or self.port != self.load_port:
+            self.load_conn = pymongo.MongoClient(host=self.load_host,
+                                                     port=self.load_port)
+        else:
+            self.load_conn = self.conn
+        self.load_collfs = gridfs.GridFS(self.load_conn[self.load_dbname], 
+                                             self.load_collname)
+        load_recent_name = '_'.join([self.load_dbname,
+                                     self.load_collname,
+                                     self.load_exp_id,
+                                     '__RECENT'])
+        self.load_collfs_recent = gridfs.GridFS(self.load_conn[load_recent_name])
 
         if cache_dir is None:
             self.cache_dir = os.path.join(os.environ['HOME'],
                                           '.tfutils',
-                                          '%s:%d' % (host, port),
-                                          dbname,
-                                          collname,
-                                          exp_id)
+                                          '%s:%d' % (self.host, self.port),
+                                          self.dbname,
+                                          self.collname,
+                                          self.exp_id)
         else:
             self.cache_dir = cache_dir
         if not os.path.isdir(self.cache_dir):
             os.makedirs(self.cache_dir)
+        
+    def load_rec(self):
+        #first try and see if anything with the save data exists, since obviously
+        #we dont' want to keep loading from the original load location if some work has
+        #already been done
+        load = self.load_from_db({'exp_id': self.exp_id},
+                                 cache_model=True)
+        #if not, try loading from the loading location
+        if not load and not self.sameloc:
```

Comment:
hm....  no I think that when no record is found, actually the return is a single None (see line 262